inf-clojure.el

```diff
@@ -208,6 +209,21 @@ whichever process buffer you want to use.")
 
 (put 'inf-clojure-mode 'mode-class 'special)
 
+(defcustom inf-clojure-repl-flavor 'clojure
```

Comment:
Basically such variable such be declared as `defvar-local` instead of as `defcustoms`. There should probably be some defcustoms with the values for the different types of repls, but the actual thing that would be set in the code would be a local variable that's specific to the REPL. You can see in CIDER's code it makes heavy use of local variables.